src/modules/flow/iio/nodes.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "iio-gen.h"
+#include "sol-flow-internal.h"
+
+#include <sol-util.h>
+#include <errno.h>
+
+#include <sol-iio.h>
+
+struct gyroscope_data {
+    struct sol_flow_node *node;
+    struct sol_iio_device *device;
+    struct sol_iio_channel *channel_x;
+    struct sol_iio_channel *channel_y;
+    struct sol_iio_channel *channel_z;
+    bool buffer_enabled : 1;
+};
+
+static void
+reader_cb(struct sol_iio_device *device, struct sol_iio_buffer *buffer, void *data)
+{
+    const char errmsg[] = "Could not read channel buffer values";
```

Comment:
When you declare as `errmsg[]`, it's stored in the `.rodata` section and then (if it's not static) copied from there to the call frame every time this function is called. A variable being `const` doesn't really make it read-only/constant: it's just an instruction for the compiler to warn/fail if one tries to write to it without removing the const with a cast.

If it's declared as `*errmsg`, however, it's not copied from the `.rodata` section; it merely points there. The const here is to the pointer, not the pointee.

Since this variable is not exported, the compiler will most likely treat this as `const char * const errmsg = "foo"`: both the pointer and the pointee will be stored in the `.rodata` section. Using `static` will not help much but it's another hint that this won't change.